src/rshim.c

```diff
@@ -2157,30 +2203,183 @@ static int rshim_update_locked_mode(rshim_backend_t *bd)
       int rt;
 
       rshim_lock();
-      pthread_mutex_lock(&bd->mutex);
       rt = rshim_access_check(bd); 
-      pthread_mutex_unlock(&bd->mutex);
       rshim_unlock();
       if (rt) {
         RSHIM_INFO("rshim%d attached by another device. Entering Drop Mode\n",
             bd->index);
+        pthread_mutex_lock(&bd->mutex);
         rshim_set_drop_mode(bd, 1);
+        pthread_mutex_unlock(&bd->mutex);
       }
     }
   }
 
   return 0;
 }
 
+static int read_rshim_sp1(rshim_backend_t *bd, uint64_t* sp1)
```

Comment:
Can we use existing naming style, i.e. rshim_read_sp1()...  Same as other new functions below.